lib/tether/ops_linux.go

```diff
@@ -442,20 +438,115 @@ func apply(t Netlink, endpoint *metadata.NetworkEndpoint) error {
 			}
 			log.Infof("Added nameserver: %s", server.String())
 		}
-	} else {
+	} else if !ip.IsUnspecifiedIP(endpoint.Network.Gateway.IP) {
```

Comment:
For endpoints connected to the bridge network, it will be our internal DNS (i.e. vch). For DHCP endpoints, which should never be the same as the bridge endpoint, it most likely will not go into this branch as we will use the DNS entries from the DHCP server.